zucchini-ui-backend/src/main/java/io/zucchiniui/backend/scenario/views/ScenarioViewAccess.java

```diff
@@ -73,6 +75,31 @@ public ScenarioViewAccess(final ScenarioDAO scenarioDAO, final TestRunRepository
         return MorphiaUtils.streamQuery(query).collect(Collectors.toMap(Scenario::getScenarioKey, scenarioToListItemViewMapper::map));
     }
 
+    public List<GroupedFailuresListItemView> getGroupedFailedScenarii(final Consumer<ScenarioQuery> preparator) {
+        List<GroupedFailuresListItemView> groupedFailures = new ArrayList<>();
+        getFailedScenarii(preparator).forEach(scenario -> {
+            boolean matchFound = false;
+            for (GroupedFailuresListItemView current : groupedFailures) {
+                if (ErrorMessageGroupingUtils.isSimilar(current.getErrorMessage(), scenario.getErrorMessage())) {
+                    current.addFailedScenario(scenario);
+                    matchFound = true;
+                    break;
+                }
+            }
+            if (!matchFound) {
+                GroupedFailuresListItemView noMatch = new GroupedFailuresListItemView();
+                noMatch.setErrorMessage(scenario.getErrorMessage());
+                noMatch.setFailedScenarii(new TreeSet<>(Comparator.comparing(failure -> failure.getInfo().getName())));
+                noMatch.addFailedScenario(scenario);
+                groupedFailures.add(noMatch);
+            }
+        });
+        return groupedFailures
+            .stream()
+            .sorted(Collections.reverseOrder(Comparator.comparingInt(grp -> grp.getFailedScenarii().size())))
```

Comment:
Simpler :

`Collections.reverseOrder(Comparator.comparingInt(grp -> grp.getFailedScenarii().size()))`

To :

`Comparator.comparingInt(grp -> grp.getFailedScenarii().size()).reversed()`